tools/vm-migrator/instance.py

```diff
@@ -0,0 +1,397 @@
+                                     delete_operation['name'])
+            return name
+        else:
+            raise Exception(
+                "Cant Delete the instance as machine image not found")
+    except Exception as ex:
+        logging.error(ex)
+        print(ex)
+        raise ex
+
+
+def get_region_from_zone(zone):
+    match = re.search(r"(\w+)-(\w+)-(\w+)", zone)
+    if (len(match.groups()) != 3):
+        raise Exception('Invalid Zone Format')
+    return match.group(1) + "-" + match.group(2)
+
+
+def get_ip(compute, project, address, region):
+    result = compute.addresses().get(project=project,
+                                     region=region,
+                                     address=address).execute()
+    return result['address']
+
+
+def reserve_internal_ip(compute, project, name, region, subnet, ip):
+    config = {
+        "address": ip,
+        "addressType": "INTERNAL",
+        "name": name,
+        "subnetwork": subnet
+    }
+    logging.info(("Reserving internal ip %s and %s") % (name, ip))
+    insert_operation = compute.addresses().insert(project=project,
+                                                  region=region,
+                                                  body=config).execute()
+    wait_for_regional_operation(compute, project, region,
+                                insert_operation['name'])
+
+    return insert_operation['selfLink']
+
+
+def upgrade_machine_type(machine_type, destination_zone):
+
+    #  the machine type is of the form
+    # https://www.googleapis.com/compute/beta/projects/pso-suchit/zones/us-east1-c/machineTypes/n1-standard-4
+    logging.info('looking for upgrading the machine type')
+    if (machine_type.startswith('projects')):
+        machine_type = '/' + machine_type
+
+    response = re.search(
+        r'\/projects\/(.*?)\/zones\/(.*?)\/machineTypes\/(.*?)$', machine_type)
+    if (len(response.groups()) != 3):
+        raise Exception('Invalid Machine Type Format')
+
+    original_machine_type = response.group(3)
+    original_zone = response.group(2)
+    destination_machine_type = machine_type_mapping.find.get(
+        original_machine_type)
+    if (destination_machine_type):
+        logging.info('found a match to upgrade the machine type %s with %s' %
+                     (machine_type, destination_machine_type))
+        machine_type = machine_type.replace(original_machine_type,
+                                            destination_machine_type)
+        machine_type = machine_type.replace(original_zone, destination_zone)
+        logging.info("upgrading machine to %s" % (machine_type))
+        return machine_type
+
+    # could not find the mapping so dont change the machine type
+    return None
+
+
+def create_instance(compute, project, zone, network, subnet, name,
+                    alias_ip_ranges, node_group, disk_names, ip, machine_type):
+    """
+    Create Instance method create a new GCP VM from the machine image
+    """
+
```

Comment:
Remove blank line (no blank lines after docstring). Also, add a dot at the end of the docstring.

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings